api_app/analyzers_manager/observable_analyzers/koodous.py

```diff
@@ -0,0 +1,59 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class Koodous(classes.ObservableAnalyzer):
+    base_url: str = "https://developer.koodous.com/apks/"
+    query_analysis = "/analysis"
+
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        try:
+            response_first = requests.request(
+                "GET",
+                self.base_url + self.observable_name,
+                headers={"Authorization": f"Token {self.__api_key}"},
+                data={},
```

Comment:
Honestly I don't get why you can't use classic `requests.get`. I am quite sure there something wrong in the test you are doing.

Reply:
`response_ = requests.get('https://developer.koodous.com/apks/5b14f214b51768a53a10fcb3632a1165de1796591cd53d077618199da48ef823/analysis',params=headers)`

this is the code i am using to test the API call. The output is like this
<Response [401]>
{'detail': 'Authentication credentials were not provided.'}

as soon as I remove '/analysis' it runs fine giving me the desired response. The problem is occurring for analysis query

Reply:
you are setting the headers in the parameters... `params=headers`